code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/InputBoxNotificationBuilderTest.kt

```diff
@@ -0,0 +1,180 @@
+
+        assertEquals(1, notificationBuilder.mActions.size)
+        assertNotNull(notificationBuilder)
+    }
+
+    @Test
+    fun `createInputReceivedPendingIntent should not set class when trackerActivityClass parameter is null`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData()
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            null,
+            broadcastReceiverClass
+        )
+
+        val notification = notificationBuilder.build()
+        val pendingIntent = notification.contentIntent
+        val shadowPendingIntent = Shadows.shadowOf(pendingIntent)
+        val intent = shadowPendingIntent.savedIntent
+
+        assertNotNull(intent)
+        assertNull(intent.resolveActivity(context.packageManager))
+    }
+
+    @Test
+    fun `createInputReceivedPendingIntent should set class when trackerActivityClass parameter is null`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData()
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        val notification = notificationBuilder.build()
+        val pendingIntent = notification.contentIntent
+        val shadowPendingIntent = Shadows.shadowOf(pendingIntent)
+        val intent = shadowPendingIntent.savedIntent
+
+        assertNotNull(intent)
+        assertEquals("com.adobe.marketing.mobile.notificationbuilder.internal.builders.DummyActivity", intent.resolveActivity(context.packageManager).className)
+    }
+
+    @Test
+    fun `Action with default hint text should be created when inputTextHint field is empty`() {
+        val dataMap = MockInputBoxPushTemplateDataProvider.getMockedInputBoxDataMapWithRequiredData()
+        dataMap[PushTemplateConstants.PushPayloadKeys.INPUT_BOX_HINT] = ""
+
+        val pushTemplate = InputBoxPushTemplate(MapData(dataMap))
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        val actions = notificationBuilder.mActions
+        assertEquals(PushTemplateConstants.DefaultValues.INPUT_BOX_DEFAULT_REPLY_TEXT, actions[0].title)
+    }
+
+    @Test
+    fun `Action with default hint text should be created when inputTextHint field is null`() {
+        val dataMap = MockInputBoxPushTemplateDataProvider.getMockedInputBoxDataMapWithRequiredData()
+        dataMap.remove(PushTemplateConstants.PushPayloadKeys.INPUT_BOX_HINT)
+
+        val pushTemplate = InputBoxPushTemplate(MapData(dataMap))
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
```

Comment:
can we verify that notificationBuilder is not null before verifying the actions?